### PR TITLE
Remove thread limiting.

### PR DESCRIPTION
**Pull request by KochetovNicolai**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix unnecessary limiting for the number of threads for selects from local replica.

### CODE REVIEW

**Comment by KochetovNicolai** [comment]
@azat I have already forgotten why do we limit threads here.
I suppose that this is not needed already, because the number of threads is limited by pipeline itself (it shouldn't be more than max streams for every step).

**Comment by alexey-milovidov** [comment]
Perf test? (reading small amount of data from a table with large amount of parts)

**Comment by KochetovNicolai** [comment]
Sure

**Comment by azat** [comment]
>@azat I have already forgotten why do we limit threads here.

It had been added in #11114

Quote from the comment to the patch from that PR:
```
Before this patch pipeline.setMaxThreads() was called with 0 (since
streams.size() == 0, while pipes.size() == 1), which leads that
pipeline.getMaxThreads() returns max_streams.
```

>I suppose that this is not needed already, because the number of threads is limited by pipeline itself (it shouldn't be more than max streams for every step).

Yes, since there is no more deprecated streams (non-processors) in this code.

>Perf test? (reading small amount of data from a table with large amount of parts)

@alexey-milovidov FWIW there is a test for this `tests/queries/0_stateless/01283_max_threads_simple_query_optimization.sql`

File: tests/performance/local_replica.xml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<test>
+    <query>select sum(number) from remote('127.0.0.{{1|2}}', numbers_mt(1000000000)) group by bitAnd(number, 1)</query>
```
Comment (azat) [inline tests/performance/local_replica.xml:2]: Hm, not sure that this query covers that case, since this reads lots of rows and also it uses numbers_mt

Comment (KochetovNicolai) [reply tests/performance/local_replica.xml:2]: Right in this case the number of threads was limited to 1.

**Comment by KochetovNicolai** [comment]
@azat tests are ok.
What do you think, do we have other reasons to keep this code?

**Comment by azat** [comment]
>@azat tests are ok.
>What do you think, do we have other reasons to keep this code?

Nope, LGTM

**Pull request merged**